File: app/models/item.rb
class Item
  include Mongoid::Document
  include Mongoid::Timestamps
  include Mongoid::Enum

  default_scope ->{ order(name: :asc) }

  field :name, type: String
  field :price, type: BigDecimal

  enum :item_type, [:broadsheet, :tabloid], default: :broadsheet

  validates_presence_of :name, :price
  validates_uniqueness_of :name

  index({ name: 1 }, { unique: true })
end

File: client/src/app/models/item.ts
import { MongoidId } from './mongoid-id';
import { Timestamps } from './timestamps';

export interface ItemParams {
  name: string;
  price: number;
  _item_type: 'broadsheet' | 'tabloid';
}

export interface Item extends ItemParams, MongoidId, Timestamps {}
